Roles (#48)
plantanapp/documentation: docs/Roles.md

<iframe width="560" height="315" src="https://www.youtube.com/embed/zOGVh-zEsqs" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

## `Permision Scheme`

### `Pages`
Setting page permission will restrict access to the whole content of a page. Also edit permission can be enabled based on the roles. 

### `Users`
Each role can be assigned or removed from a user via the users page in the configuration page of Plant an App.

### `Entities`
Permissions set on a entity level will automatically change the permissions of the autogenerated page and the access of what operations can be done via the user interface like Create Read Update or Delete. Also the same permissions will be applied to the actions witch are used to do CRUD operations like Create, Update, Partially Update, Read and Delete an Entity.

### `Modules`
Each Module individually has each own permissions and can be modified individually to restrict access or grant edit options to roles. 

### `Actions & Buttons`

> All the actions and buttons that can be used in the Plant an App platform have 2 settings called Condition (Show Condition) and Enable Condition that supports tokens. Based on those we can create complex access securities with the HasRole token.  `[HasRole:RoleName]` will check if the logged in user (or the user from [Load User](/docs/Actions/load-user.md)/[Load Users from SQL](/docs/Actions/load-users-sql.md)) and return `true` or `false` depending if the user it is in that role or not. (eg: [HasRole:Admin] ==  true can be used on a button to only enable it for an user in the Admin role)

Reviewer: ```suggestion
All the actions and buttons that can be used in the Plant an App platform have 2 settings called Condition (Show Condition) and Enable Condition that supports tokens. Based on those we can create complex access securities with the HasRole token.  `[HasRole:RoleName]` will check the logged-in user (or the user from [Load User](/docs/Actions/load-user.md)/[Load Users from SQL](/docs/Actions/load-users-sql.md)) and return `true` or `false` depending if the user is in that role or not. (eg: [HasRole:Admin] ==  true can be used on a button to only enable it for a user in the Admin role)

```

### `API JWT Autentification`

Authenticating with JWT to use an API resources has to be done in the name of a user so permissions can be set to only permit access to specific users in those roles.

### `Search Index Rules`

> All or some results may restricted (not shown) based on the roles by specifying the roles who have access to each results.

Reviewer: ```suggestion
All or some results may be restricted (not shown) based on the roles by specifying the roles who have access to each result.

```

## `Role Management`

### `Adding Roles`

Click on the `NEW` button on the top right of the page to start adding a new role. Input the Role Name and Role Description. Click Save to add the new role.

### `Editing Roles`

In the user grid find the role you want to edit and click in the edit icon to open the edit dialog. Modify the Role Name and/or Role Description. Click Save to modify the role.

### `Removing  Roles`

In the user grid find the role you want to delete and click in the delete icon (trash can) to open the delete dialog. Click Yes to delete the role.

## `Built-In Roles`

### `Admin`

> It's indented for users that need to manage the application setting and can also manage users in the Built-In roles. People in this role are involved in the development of the application. This role can't be edited or deleted.

Reviewer: ```suggestion
It's intended for users that need to manage the application settings and can also manage users in the Built-In roles. People in this role are involved in the development of the application. This role can't be edited or deleted.

```
